### ISSUE TITLE
API key and org id security.

### ISSUE THREAD

**Issue by Reed-Broadhead** [Reed-Broadhead/VC-Connect#4]
**Is your feature request related to a problem? Please describe.**
We need a way to use both the API key and org ID in the create meeting post request without compromising them. Post a comment if you have a solution you thing might work.
this issue is dependent on both #2 and #3.

**Describe the solution you'd like**
No user can have access to either the org ID or API key value.

**Describe alternatives you've considered**

**Additional context**

**Comment by Reed-Broadhead** [comment]
I think whenever our users need to make a call to Dyte that requires sensitive information we should first send the request to our own API which will store our API key and org id and make the call to Dyte for us. if we use this solution we could prevent the API key and org id from ever being seen by the front end. 

What do you guys think about this solution, Are there any problems that I don't see?

**Comment by MichaelJHTaggart** [comment]
@Reed-Broadhead  and I looked into an alternative option; Using environment variables.

TLDR; It gets directly exposed to the front end.

Here we see that the env variable we created is being directly exposed:
![Screenshot_2024-02-20_231629](https://github.com/Reed-Broadhead/VC-Connect/assets/63270278/8fd10069-03f0-410b-93f4-809f0f0b6859)

Here is some else also mentioning this same issue:
[https://forum.freecodecamp.org/t/keeping-api-key-hidden-using-react/272539](https://forum.freecodecamp.org/t/keeping-api-key-hidden-using-react/272539)

Here is the official documentation also advising against this: 

1. [https://forum.freecodecamp.org/t/keeping-api-key-hidden-using-react/272539](https://forum.freecodecamp.org/t/keeping-api-key-hidden-using-react/272539)
2. [https://vitejs.dev/guide/env-and-mode](https://vitejs.dev/guide/env-and-mode)

**Comment by MichaelJHTaggart** [comment]
@Reed-Broadhead added a back-end server to solve that No user can have access to either the org ID or API key value.

Next issue:

- [ ] How can Reed (or any of us) share sensitive API Keys to each other securely?
- [ ] Quick Readme of how to use nodemon/server

**Comment by MichaelJHTaggart** [comment]
Because no ones credit card is tied to the API Key to Dyte. We have zero risk of "damages" occurring to anyone, so per @Reed-Broadhead idea we could just share this in Discord. 

I did find some great articles that illustrate some important things to consider when you do have stakes or risk of actual money or reputation getting used/destroyed:

Considerations for sharing api keys:

1. Risk of loss of confidentiality of the key. Consider security of both transfer and storage.
2. API key should provide least privilege permissions
3. Can I tell who or what is using the api key? If you share one key among many people and several systems, that will make it harder to attribute actions take with the key to a specific person or system. Logging ip addresses using the key helps mitigate this.
4. Company policy or other guidance on the issue

Some transfer options:

1. Use a password manager like bitwarden, dashlane, 1password and put the api key in a vault you share with others as needed. (*This takes care of security & storage of the data)
2. Put the key in the native vault of your CI/CD system if that’s where it will be used
3. Put the key in the native secret manager of whatever cloud provider you use if a system will pull the key from there
4. If the key needs to be in a database, put it there
